models/provider.go

```diff
@@ -13,3 +13,25 @@ type Provider struct {
 	Telecoms       []Telecom       `json:"telecoms,omitempty"`
 	CDAIdentifiers []CDAIdentifier `json:"cda_identifiers,omitempty"`
 }
+
+// IdentifierForRoot takes a Provider and a string identifying the root OID, and returns the extension for that OID
+func (prov *Provider) IdentifierForRoot(root string) string {
+	for _, identifier := range prov.CDAIdentifiers {
+		if identifier.Root == root {
+			return identifier.Extension
+		}
+	}
+	return ""
+}
+
+// IdentifiersForRoot applies IdentifierForRoot to a passed-in slice of Providers
+func IdentifiersForRoot(prov []Provider, root string) []string {
+	ids := []string{}
+	for _, provider := range prov {
+		str := provider.IdentifierForRoot(root)
+		if str != "" {
```

Comment:
Minor nitpicks but can you make the assignment to str an initialization statement for the if? And maybe rename `str` to `rootID` or `id`, and `prov` to `provs`?